### PR TITLE
KEYCLOAK-5938 Authentication sessions: Support for logins of multiple…

### PR DESCRIPTION
**Pull request by mposolda**
… tabs of same client

### CODE REVIEW

File: services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java [hunk 1]
```diff
@@ -215,22 +216,41 @@ public static void backchannelLogout(KeycloakSession session, RealmModel realm,
...
+        }
+
+        // See if we have logoutAuthSession inside current rootSession. Create new if not
+        Optional<AuthenticationSessionModel> found = rootLogoutSession.getAuthenticationSessions().values().stream().filter((AuthenticationSessionModel authSession) -> {
+
+            return client.equals(authSession.getClient()) && Objects.equals(AuthenticationSessionModel.Action.LOGGING_OUT.name(), authSession.getAction());
+
+        }).findFirst();
+
+        AuthenticationSessionModel logoutAuthSession = found.isPresent() ? found.get() : rootLogoutSession.createAuthenticationSession(client);
```
Comment (hmlnarik) [inline services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java]: nit: can be replaced with
```java
   AuthenticationSessionModel logoutAuthSession = found.orElse(rootLogoutSession.createAuthenticationSession(client));
```

**Review by hmlnarik — APPROVED** [review]
One small nit if you like to fix it, but can be merged as is.

Thanks for the changes!

File: services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java (hunk 1 above)
Comment (mposolda) [reply services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java]: Thanks Hynek for the review! For this one, I remember I wanted to do something like this when I was coding it, but it requires "rootLogoutSession" to be declared as "final" variable, which it isn't, so would need to be re-declared. So did it the current way in the end. Will doublecheck again and possibly change it...

Comment (mposolda) [reply services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java]: @hmlnarik Sorry, just tried that it works when I do it exactly the way you mentioned. I've used a bit different syntax before, which requires "rootLogoutSession" to be final :) Changed it in the other PR #4837 .

**Pull request merged**